src/overlays/fbdemos/ovl_fbdemo_wipe5/z_fbdemo_wipe5.c

```diff
@@ -16,37 +18,133 @@ void TransitionWipe5_SetColor(void* thisx, u32 color);
 void TransitionWipe5_SetEnvColor(void* thisx, u32 color);
 s32 TransitionWipe5_IsDone(void* thisx);
 
-#if 0
 const TransitionInit TransitionWipe5_InitVars = {
-    TransitionWipe5_Init,
-    TransitionWipe5_Destroy,
-    TransitionWipe5_Update,
-    TransitionWipe5_Draw,
-    TransitionWipe5_Start,
-    TransitionWipe5_SetType,
-    TransitionWipe5_SetColor,
-    TransitionWipe5_SetEnvColor,
-    TransitionWipe5_IsDone,
+    TransitionWipe5_Init,     TransitionWipe5_Destroy,     TransitionWipe5_Update,
+    TransitionWipe5_Draw,     TransitionWipe5_Start,       TransitionWipe5_SetType,
+    TransitionWipe5_SetColor, TransitionWipe5_SetEnvColor, TransitionWipe5_IsDone,
 };
 
-#endif
+void TransitionWipe5_Start(void* thisx) {
+    TransitionWipe5* this = THIS;
+
+    this->isDone = false;
+    switch (this->unk_12) {
+        case 0:
+            this->unk_8 = 0.02f;
+            break;
+        case 1:
+            this->unk_8 = 0.05f;
+            break;
+        default:
+            this->unk_8 = 0.2f;
+            break;
+    }
+    switch (this->unk_11) {
+        case 0:
+            this->primColor.rgba = 0x000000FF;
+            break;
+        case 1:
+            this->primColor.rgba = 0xA0A0A0FF;
+            break;
+        default:
+            this->primColor.rgba = 0xA0A0A0FF;
+            break;
+    }
+}
+
+void* TransitionWipe5_Init(void* thisx) {
+    TransitionWipe5* this = THIS;
+
+    bzero(this, sizeof(TransitionWipe5));
+    return this;
+}
+
+void TransitionWipe5_Destroy(void* thisx) {
+}
+
+void TransitionWipe5_Update(void* thisx, s32 updateRate) {
+    TransitionWipe5* this = THIS;
+
+    if (!this->unk_10) {
+        this->isDone = true;
+    } else {
+        this->unk_C = this->unk_C + ((this->unk_8 * 3.0f) / updateRate);
+        if (this->unk_C >= 1.0f) {
+            this->unk_C = 1.0f;
+            this->isDone = true;
+        }
+    }
+}
+
+void TransitionWipe5_Draw(void* thisx, Gfx** gfxP) {
+    Gfx* gfx = *gfxP;
+    TransitionWipe5* this = THIS;
+    s32 alpha;
+    s32 sp50 = D_801FBB90;
+    s32 width;
+    s32 height;
 
-extern u16 D_0F000000[];
+    alpha = ((1.0f - this->unk_C) * 255.0f);
+    width = gScreenWidth;
+    height = gScreenHeight;
```

Comment:
You can probably define all of these where they're declared.

Reply:
Doesn't match
https://decomp.me/scratch/2vEwW

Reply:
https://decomp.me/scratch/aMY6k
You can muck about with the stack a lot to get it to work.